packages/test/test-version-utils/src/compatUtils.ts

```diff
@@ -42,6 +47,68 @@ import {
  */
 export const TestDataObjectType = "@fluid-example/test-dataStore";
 
+/**
+ * This function modifies container runtime options according to a version of runtime used.
+ * If a version of runtime does not support some options, they are removed.
+ * If a version runtime supports some options, such options are enabled to increase a chance of
+ * hitting feature set controlled by such options, and thus increase chances of finding product bugs.
+ * 
+ * @param version - a version of container runtime to be used in test
+ * @param optionsArg - input runtime options (optional)
+ * @returns - runtime options that should be used with a given version of container runtime
+ * @internal
+ */
+function filterRuntimeOptionsForVersion(
+	version: string,
+	optionsArg: IContainerRuntimeOptions = {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				...DefaultSummaryConfiguration,
+				...{
+					initialSummarizerDelayMs: 0,
+				},
+			},
+		},
+	},
+) {
+	let options = { ...optionsArg };
+
+	// No test fails with this option, it allows us to validate properly expectations and
+	// implementation of services
+	options.loadSequenceNumberVerification = "close";
+
+	if (version === "1.3.7") {
+		options.compressionOptions = undefined;
+		options.enableGroupedBatching = false;
+		options.enableRuntimeIdCompressor = "off";
+		options.maxBatchSizeInBytes = undefined;
+		options.chunkSizeInBytes = undefined;
+	} else if (version.includes("2.0.0-rc")) {
+		const {
+			compressionOptions = {
+				minimumBatchSizeInBytes: 500,
+				compressionAlgorithm: CompressionAlgorithms.lz4,
+			},
+			chunkSizeInBytes = 200,
+			enableRuntimeIdCompressor = "on",
+			enableGroupedBatching = true,
+		} = options;
+
+		// Note about comments below:
```

Comment:
Preserving data for historical reasons (as I'll update it with recent findings soon, after merging main):

As of commit https://github.com/microsoft/FluidFramework/pull/20109/commits/9b70e49bb3d7a3fd8504a5e190f392b7f6b1b4e5 made on  Mar 3, the following was true:
```ts
		options = {
			... options,
			compressionOptions, // 2 same-version tests failed; 2 cross-version tests fail; seems like functional bugs.
			chunkSizeInBytes, // same-version test "can send and receive multiple batch ops that are flushed on JS turn" fails; 0 cross-version tests failed.
			enableRuntimeIdCompressor, // 9 same-version tests failed; 0 cross-version tests failed.
			enableGroupedBatching, // 0 same-version tests failed; 14 cross-version tests failed, all look like functional bugs.
		};

... (comment in getVersionedTestObjectProviderFromApis)

			// If you isolate each runtime option and test them individually then only 13 test fails.
			// But enabling all 2.0 options results in 69 tests failing!
			// Most of the failures are due to error generated as result of sending ops in disconnected state, which is a bit weird.
			
```